sdk/src/settings.rs

```diff
@@ -218,40 +233,327 @@ impl Default for Verify {
     builder: Builder,
+
+    #[serde(skip)]
+    config: Config,
 }
 
 impl Settings {
-    #[allow(unused)]
+    /// Returns the current [Settings] for the local thread.
+    pub fn thread_local() -> Option<Settings> {
+        // TODO: return error instead of option
+        SETTINGS
+            .with_borrow(|config| config.clone().try_deserialize::<Settings>())
+            .ok()
+    }
+
+    /// Construct a new [Settings] with default values.
+    ///
+    /// This can be used with [Setting::set_thread_local] to reset the thread local [Settings]
+    /// to their default values.
+    pub fn new() -> Settings {
+        Self::default()
+    }
+
+    /// Construct a [Settings] from a given toml string.
+    pub fn from_toml(settings_toml: &str) -> Result<Settings> {
+        let config = Config::builder()
+            .add_source(config::File::from_str(settings_toml, FileFormat::Toml))
+            .build()
+            .map_err(|_e| Error::BadParam("could not parse configuration file".into()))?;
+
+        let mut settings = config.clone().try_deserialize::<Settings>().map_err(|_e| {
+            Error::BadParam("configuration file contains unrecognized param".into())
+        })?;
+        settings.validate()?;
+
+        settings.config = config;
+
+        Ok(settings)
+    }
+
+    /// Construct [Settings] from a given toml file.
+    #[cfg(feature = "file_io")]
+    pub fn from_toml_file<P: AsRef<Path>>(setting_path: P) -> Result<Settings> {
+        let setting_buf = std::fs::read(&setting_path).map_err(Error::IoError)?;
+        Settings::from_toml(
+            &String::from_utf8(setting_buf)
+                .map_err(|_| Error::BadParam("invalid utf-8".to_owned()))?,
+        )
+    }
+
+    /// Merges the current [Settings] with thread local [Settings].
+    ///
+    /// Only fields that are present in the current [Settings] will override the fields
+    /// in the thread local [Settings].
+    pub fn set_thread_local(self) -> Result<()> {
+        let update_config = SETTINGS.with_borrow(|current_settings| {
+            Config::builder()
+                .add_source(current_settings.clone())
+                .add_source(self.config)
+                .build() // merge overrides, allows for partial changes
+        });
+
+        match update_config {
+            Ok(update_config) => {
+                SETTINGS.set(update_config);
+
+                Ok(())
+            }
+            Err(_) => Err(Error::OtherError("could not update configuration".into())),
+        }
+    }
+
+    /// Constructs a signer from the specified `trust.signer_info` in the settings.
+    ///
+    /// The returned signer can be passed to [Builder::sign][crate::Builder::sign]
+    /// and other related functions.
+    ///
+    /// This function will error with [Error::UnspecifiedSignerSettings][crate::Error::UnspecifiedSignerSettings]
+    /// if the `trust.signer_info` is unspecified.
+    pub fn signer(&self) -> Result<Box<dyn Signer>> {
```

Comment:
Why is this part of settings?  Settings do work they hold values for other code that does the actual work.